packages/shared/lib/typings/events.ts

```diff
@@ -111,38 +111,65 @@ export interface ReattachmentEventPayload {
 }
 
 export enum TransferProgressEventType {
-    // Syncing account.
+    /// Syncing account.
     SyncingAccount = 'SyncingAccount',
     /// Performing input selection.
     SelectingInputs = 'SelectingInputs',
-    /// Generating remainder value deposit address.
+    /// Generating address for remainder funds.
     GeneratingRemainderDepositAddress = 'GeneratingRemainderDepositAddress',
+    /// Preparing the transaction data.
+    PreparedTransaction = 'PreparedTransaction',
     /// Signing the transaction.
     SigningTransaction = 'SigningTransaction',
     /// Performing PoW.
     PerformingPoW = 'PerformingPoW',
     /// Broadcasting.
     Broadcasting = 'Broadcasting',
+    /// Complete.
+    Complete = 'Complete'
+}
+
+export interface TransferState {
+    /// Transfer progress event type.
+    type: TransferProgressEventType
+    /// Relevant data for this type of transfer progress event.
+    data?: any
```

Comment:
Don't we have a specific type for `data` here?

Reply:
It's has a few possible values, but I'll put an appropriate union type.